pkg/webhooks/podimagespec/podimagespec.go

```diff
@@ -0,0 +1,281 @@
+)
+
+var (
+	timeout int32 = 2
+	scope         = admissionregv1.NamespacedScope
+	rules         = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{
+				admissionregv1.Create,
+			},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{""},
+				APIVersions: []string{"v1"},
+				Resources:   []string{"pods"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log        = logf.Log.WithName(WebhookName)
+	imageRegex = regexp.MustCompile(`(image-registry\.openshift-image-registry\.svc:5000\/)(?P<namespace>\S*)(/)(?P<image>\S*)(:)(?P<tag>\S*)`)
+)
+
+// PodImageSpecWebhook mutates an image spec in a pod
+type PodImageSpecWebhook struct {
+	s          *runtime.Scheme
+	configV1   configv1.Config
+	imageV1    imagestreamv1.Image
+	kubeClient client.Client
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *PodImageSpecWebhook {
+	scheme := runtime.NewScheme()
+	err := configv1.Install(scheme)
+	if err != nil {
+		fmt.Println("Error:", err)
+		return nil
+	}
+	return &PodImageSpecWebhook{
+		s: scheme,
+	}
+}
+
+// CheckImageRegistryStatus checks the status of the image registry service
+func (s *PodImageSpecWebhook) CheckImageRegistryStatus(ctx context.Context) (bool, error) {
+	var err error
+	if s.kubeClient == nil {
+		s.kubeClient, err = k8sutil.KubeClient(s.s)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	err = s.kubeClient.Get(ctx, client.ObjectKey{Name: "cluster"}, &s.configV1)
+	if err != nil {
+		return false, fmt.Errorf("failed to get image registry config: %v", err)
+	}
+
+	// if image registry is set to managed then it is operational
+	if s.configV1.Spec.ManagementState == operatorv1.Managed {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *PodImageSpecWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	//Implement authorized next
+	return s.authorizeOrMutate(request)
+}
+
+// authorizeOrMutate decides whether the Request requires mutation before it's allowed to proceed.
+func (s *PodImageSpecWebhook) authorizeOrMutate(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+	pod, err := s.renderPod(request)
+	if err != nil {
+		log.Error(err, "Couldn't render a Pod from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	for i := range pod.Spec.Containers {
+		container := &pod.Spec.Containers[i]
+		ret, err := s.authorizeOrMutateContainer(container)
+		ret.UID = request.AdmissionRequest.UID
+		if err != nil {
+			return admissionctl.Errored(http.StatusInternalServerError, err)
+		}
+	}
+
+	for i := range pod.Spec.InitContainers {
+		container := &pod.Spec.InitContainers[i]
+		ret, err := s.authorizeOrMutateContainer(container)
+		ret.UID = request.AdmissionRequest.UID
+		if err != nil {
+			return admissionctl.Errored(http.StatusInternalServerError, err)
+		}
+	}
```

Comment:
You can only admit or reject the whole pod.. I'm not sure I understand why we are looping over multiple containers at this step? You're also overwriting the `ret` with each loop, so it will only action the last iteration of the loop.